cgmes/cgmes-model/src/main/java/com/powsybl/cgmes/model/CgmesOnDataSource.java

```diff
@@ -32,29 +34,55 @@ public ReadOnlyDataSource dataSource() {
         return dataSource;
     }
 
+    private boolean checkIfMainFileNotWithCgmesData(boolean isCim14) {
+        if (dataSource.getDataExtension() == null || dataSource.getDataExtension().isEmpty()) {
+            return false;
+        } else if (EXTENSIONS.equals(dataSource.getDataExtension())) {
+            try (InputStream is = dataSource.newInputStream(null, EXTENSIONS)) {
```

Comment:
Does this do an NPE if the user specified a data extension but the file doesn't actually exist ?

Reply:
It would throw an exception but not a NPE. It will be fixed in https://github.com/powsybl/powsybl-core/pull/3114